test/fileutils/test_fileutils.rb

```diff
@@ -97,8 +97,12 @@ def root_in_posix?
       if /cygwin/ =~ RUBY_PLATFORM
         # FIXME: privilege if groups include root user?
         return Process.groups.include?(0)
-      end
-      if Process.respond_to?('uid')
+      elsif system('command', '-v', 'capsh', out: File::NULL)
+        return system(
+          'capsh', '--has-p=CAP_DAC_OVERRIDE', '--has-p=CAP_CHOWN', '--has-p=CAP_FOWNER',
+          out: File::NULL, err: File::NULL
+        )
+      elsif Process.respond_to?('uid')
```

Comment:
I don't think you should assume `command -v` to work as expected on non-POSIX systems.

```suggestion
      end
      begin
        return system(
          'capsh', '--has-p=CAP_DAC_OVERRIDE', '--has-p=CAP_CHOWN', '--has-p=CAP_FOWNER',
          out: File::NULL, err: File::NULL, exception: true
        )
      rescue Errno::ENOENT
      end
      if Process.respond_to?('uid')
```

Reply:
👍  - I refactored it a bit differently (so we could keep the same if... else if... structure in `root_in_posix?`). The cost is that it means `capsh` is executed twice if it's present but.... seems like not a big deal for a test.